p4rt_app/sonic/app_db_manager.cc

```diff
@@ -245,6 +245,184 @@ void WriteBatchToAppDb(
+    pdpi::IrMatchFieldDefinition ir_match = match_pair.second;
+
+    match_json["id"] = ir_match.match_field().id();
+    match_json["name"] = ir_match.match_field().name();
+    match_json["bitwidth"] = ir_match.match_field().bitwidth();
+    match_json["format"] = pdpi::Format_Name(ir_match.format());
+    nlohmann::json references_json = {};
+    for (const auto& ref_pair : ir_match.references()) {
+      nlohmann::json ref_json = nlohmann::json({});
+      ref_json["table"] = ref_pair.table();
+      ref_json["match"] = ref_pair.match_field();
+      references_json.push_back(ref_json);
+    }
+    match_json.push_back(
+         nlohmann::json::object_t::value_type("references", references_json));
+    matchfields_json.push_back(match_json);
+  }
+  table_json.push_back(
+       nlohmann::json::object_t::value_type("matchFields", matchfields_json));
+
+  nlohmann::json actions_json = {};
+  for (const auto& action : ir_table.entry_actions()) {
+    nlohmann::json action_json = nlohmann::json({});
+    pdpi::IrActionDefinition ir_action = action.action();
+
+    action_json["id"] = ir_action.preamble().id();
+    action_json["name"] = ir_action.preamble().name();
+    action_json["alias"] = ir_action.preamble().alias();
+    nlohmann::json params_json = {};
+    for (const auto& param : ir_action.params_by_name()) {
+      nlohmann::json param_json = nlohmann::json({});
+      pdpi::IrActionDefinition::IrActionParamDefinition ir_param = param.second;
+
+      param_json["id"] = ir_param.param().id();
+      param_json["name"] = ir_param.param().name();
+      param_json["bitwidth"] = ir_param.param().bitwidth();
+      param_json["format"] = pdpi::Format_Name(ir_param.format());
+      nlohmann::json references_json = {};
+      for (const auto& ref_pair : ir_param.references()) {
+        nlohmann::json ref_json = nlohmann::json({});
+        ref_json["table"] = ref_pair.table();
+        ref_json["match"] = ref_pair.match_field();
+        references_json.push_back(ref_json);
+      }
+      param_json.push_back(
+           nlohmann::json::object_t::value_type("references", references_json));
+      params_json.push_back(param_json);
+    }
+    action_json.push_back(
+          nlohmann::json::object_t::value_type("params", params_json));
+    actions_json.push_back(action_json);
+  }
+  table_json.push_back(
+       nlohmann::json::object_t::value_type("actions", actions_json));
+
+  if (ir_table.counter().unit() != p4::config::v1::CounterSpec::UNSPECIFIED) {
+    // Counter units: BYTES, PACKETS, BOTH
+    table_json.push_back(nlohmann::json::object_t::value_type("counter/unit",
+         p4::config::v1::CounterSpec::Unit_Name(ir_table.counter().unit())));
+  }
+
+  tables.push_back(table_json);
+
+  return absl::OkStatus();
+}
+
+// Publish set of tables in json formatted string to AppDb
+absl::StatusOr<std::string> PublishTablesDefinitionToAppDb(
+    const std::string& tables_info_s,
```

Comment:
nit: Keeping inline with the styling in this file, var names are generally not shortened, so maybe ext_tables_info ?